### DISCUSSION TITLE
Accessing dependencies of dependencies

### DISCUSSION THREAD

**Discussion by zaaakher** [pnpm/pnpm#7501] [Q&A]
Hi,

to summarize discussions like [this](https://github.com/davidjerleke/embla-carousel/pull/658#issuecomment-1878998416) and [this](https://github.com/shadcn-ui/ui/pull/2286#issuecomment-1879983060)

## with npm

when I install `embla-carousel-react`, which depends on `embla-carousel` I will then be able to access `embla-carsouel` and import things from it even though it's not listed in my `package.json`, and I can find `embla-carousel` listed in my `node_modules`

## with pnpm

`embla-carousel` becomes inaccessible to me and the only place I could find it is in `node_modules/.pnpm` 

---

My workaround is to explicitly install `embla-carousel` as a devDependency since I only need to access it during development.

Any thoughts on how to go about this?

**Comment by TheMikeyRoss** [comment]
cc @zkochan

**Comment by guoyunhe** [comment]
I think it is by design (I don't like it either, but here is no way to avoid it):

> The good thing about pnpm's proper node_modules structure is that it "[helps to avoid silly bugs](https://www.kochan.io/nodejs/pnpms-strictness-helps-to-avoid-silly-bugs.html)" by making it impossible to use modules that are not specified in the project's package.json.
> Source: https://pnpm.io/pnpm-vs-npm
